### ISSUE TITLE
[JOSS] Ubuntu installation error

### ISSUE THREAD

**Issue by sfmig** [FabianPlum/OmniTrax#10]
I followed the instructions in [the guide](https://github.com/FabianPlum/OmniTrax/tree/main?tab=readme-ov-file#installation-guide) for installation in Ubuntu, but unfortunately got an error during the [the test drive](https://github.com/FabianPlum/OmniTrax/tree/main?tab=readme-ov-file#a-quick-test-drive-detection--tracking)

When hitting TRACK I got:
```
OSError: /home/sminano/.config/blender/2.92/scripts/addons/omni_trax/darknet/libdarknet.so: cannot open shared object file: No such file or directory
```
Using Ubuntu 20.04 (and Blender 2.92.0).

Let me know if any more info is required!

**Comment by FabianPlum** [comment]
That's a shame! Could you send me a list of the contents of the darknet folder? Perhaps something went wrong during the installation

**Comment by sfmig** [comment]
wait I think I downloaded the omni_trax.zip file from the link in the README, which points to version 0.2.3, rather than the latest release. Will double check if this fixes it tomorrow!

To have the link in the README always point to the latest release, this link should do the trick: https://github.com/FabianPlum/OmniTrax/releases/latest/download/omni_trax.zip

**Comment by sfmig** [comment]
with version 0.2.3 this was the content of the darknet folder
<details>
  <summary>`ls -la` output</summary>
  <pre><font color="#4E9A06"><b>sminano@Precision-3660</b></font>:<font color="#3465A4"><b>~/.config/blender/2.92/scripts/addons/omni_trax/darknet</b></font>$ ls -la
  total 29228
  drwxrwxr-x 3 sminano sminano    4096 Feb 21 11:13 <font color="#3465A4"><b>.</b></font>
  drwxrwxr-x 4 sminano sminano    4096 Feb 21 11:13 <font color="#3465A4"><b>..</b></font>
  -rw-rw-r-- 1 sminano sminano 1271808 Feb 21 11:13 darknet_cpu.dll
  -rw-rw-r-- 1 sminano sminano 1266688 Feb 21 11:13 darknet_cpu.exe
  -rw-rw-r-- 1 sminano sminano   11505 Feb 21 11:13 darknet_cpu.py
  -rw-rw-r-- 1 sminano sminano 2674176 Feb 21 11:13 darknet.dll
  -rw-rw-r-- 1 sminano sminano 2374656 Feb 21 11:13 darknet.exe
  -rw-rw-r-- 1 sminano sminano   11592 Feb 21 11:13 darknet.py
  -rw-rw-r-- 1 sminano sminano  617472 Feb 21 11:13 jpeg62.dll
  -rw-rw-r-- 1 sminano sminano  195584 Feb 21 11:13 libpng16.dll
  -rw-rw-r-- 1 sminano sminano 2449408 Feb 21 11:13 libprotobuf.dll
  -rw-rw-r-- 1 sminano sminano     527 Feb 21 11:13 LICENSE
  -rw-rw-r-- 1 sminano sminano  154624 Feb 21 11:13 lzma.dll
  -rw-rw-r-- 1 sminano sminano 2139136 Feb 21 11:13 opencv_calib3d.dll
  -rw-rw-r-- 1 sminano sminano 3461632 Feb 21 11:13 opencv_core.dll
  -rw-rw-r-- 1 sminano sminano 3533824 Feb 21 11:13 opencv_dnn.dll
  -rw-rw-r-- 1 sminano sminano  772608 Feb 21 11:13 opencv_features2d.dll
  -rw-rw-r-- 1 sminano sminano  418816 Feb 21 11:13 opencv_flann.dll
  -rw-rw-r-- 1 sminano sminano  287232 Feb 21 11:13 opencv_highgui.dll
  -rw-rw-r-- 1 sminano sminano  302592 Feb 21 11:13 opencv_imgcodecs.dll
  -rw-rw-r-- 1 sminano sminano 4803072 Feb 21 11:13 opencv_imgproc.dll
  -rw-rw-r-- 1 sminano sminano  647168 Feb 21 11:13 opencv_video.dll
  -rw-rw-r-- 1 sminano sminano  540160 Feb 21 11:13 opencv_videoio.dll
  -rw-rw-r-- 1 sminano sminano  103424 Feb 21 11:13 pthreadVC3.dll
  drwxrwxr-x 2 sminano sminano    4096 Feb 21 11:15 <font color="#3465A4"><b>__pycache__</b></font>
  -rw-rw-r-- 1 sminano sminano  186368 Feb 21 11:13 swresample-3.dll
  -rw-rw-r-- 1 sminano sminano  744448 Feb 21 11:13 swscale-5.dll
  -rw-rw-r-- 1 sminano sminano  444928 Feb 21 11:13 tiff.dll
  -rw-rw-r-- 1 sminano sminano  369152 Feb 21 11:13 webp.dll
  -rw-rw-r-- 1 sminano sminano   87040 Feb 21 11:13 zlib1.dll
  </pre>
</details>

I uninstalled Blender (removed the application parent folder and the blender dir under `.config/`), reinstalled it, and tried again with version 0.3.1 of the add-on. In this case I got a `ModuleNotFoundError` error:
<pre>Writing userprefs: &apos;/home/sminano/.config/blender/2.92/config/userpref.blend&apos; ok
addon_utils.disable: omni_trax not disabled
Modules Installed (omni_trax) from &apos;/home/sminano/Downloads/omni_trax.zip&apos; into &apos;/home/sminano/.config/blender/2.92/scripts/addons&apos;

INFO: Checking requirements for omni_trax addon...


INFO: successfully loaded OmniTrax
Traceback (most recent call last):
  File &quot;/home/sminano/software/blender-2.92.0-linux64/2.92/scripts/modules/addon_utils.py&quot;, line 351, in enable
    mod = __import__(module_name)
  File &quot;/home/sminano/.config/blender/2.92/scripts/addons/omni_trax/__init__.py&quot;, line 14, in &lt;module&gt;
    import cv2
ModuleNotFoundError: No module named &apos;cv2&apos;
</pre>

It seems like it didn't install the dependencies?

Let me know if you'd like me to try anything else - in the meantime I'll try to check on a WIndows machine

**Comment by FabianPlum** [comment]
Huh, interesting. 
Did it throw any errors during the installation?
I'll grab an Ubuntu machine and see if I can replicate the error!

**Comment by FabianPlum** [comment]
Ah, I think I found the issue:

When you uninstalled blender, it will likely not have removed the associated entries in "/home/sminano/.config/blender/..." . Try removing the blender directory (or just the version directory 2.92, in case you have other versions you want to keep), and then re-running the installation process.

(EDIT: The installation process does not run correctly, see below. I'd still advice to remove the directories mentioned here, just to make sure you start with a clean install)

**Comment by FabianPlum** [comment]
Oh! Could you do me a favour and double check the contents of the "omni_trax/setup_state.txt" file? 

I think my packaging tool picked up the wrong version and set all flags to true, so blender then just assumes everything is already installed and does not proceed to build correctly.

The master repo contains the correct setup_state.txt file which should just contain:

> setup_complete=False

I'll update the packaged version. Then the installation should run correctly.

**Comment by FabianPlum** [comment]
Okay, I updated the file contained with the release:

https://github.com/FabianPlum/OmniTrax/releases/tag/V_0.3.1

Now things should install as expected.

Thanks for your patience!

**Comment by sfmig** [comment]
it worked with the updated zip! 🥳 

I would just recommend replacing the link to the zip file [here](https://github.com/FabianPlum/OmniTrax/blob/102e4743ceb5bd064e68fd641975bb48cfe2a592/README.md?plain=1#L54) with this link `https://github.com/FabianPlum/OmniTrax/releases/latest/download/omni_trax.zip`, so that it points to the latest release (right now it's fixed to the zip of v0.2.3)

**Comment by FabianPlum** [comment]
Amazing! Thanks a lot for testing!

Good spot, I thought I had replaced the static links everywhere but will fix that one as well.
